src/rov/rov_pca9685/src/ServoDriver.cpp

```diff
@@ -1,171 +1,91 @@
 #include "pca9685/ServoDriver.hpp"
 #include <iostream>
+#include <rclcpp/rclcpp.hpp>
 
 #define NUM_COUNTS 4177
+#define BUS_DEV 1
+#define ADDRESS 0x41
 
 static int f2imap(float value, float from_min, float from_max, int to_min, int to_max){
-    return ((value - from_min) * ((to_max - to_min) / (from_max - from_min)) + to_min);
+    int val = ((value - from_min) * ((to_max - to_min) / (from_max - from_min)) + to_min);
+    val = val < to_min ? to_min : val;
+    val = val > to_max ? to_max : val;
+    return val;
 }
 
-ServoDriver::ServoDriver(){
-    this->driver_board = PCA9685(0x41);
-    if(!this->driver_board.openPCA9685()) {
+ServoDriver::ServoDriver() {
+    this->driver_board = PCA9685(BUS_DEV, ADDRESS);
+    if(!this->driver_board.Open()) {
         throw std::runtime_error("Failed to open PCA9685");
     }
-    this->driver_board.setAllPWM(0,0);
-    this->driver_board.reset();
-    this->setPWMFrequency(this->driver_board_frequency);
-}
-
-ServoDriver::~ServoDriver(){}
-
-void ServoDriver::addServo(int channel){
-    if(this->isChannelNotInUse(channel)){
-        Servo new_servo;
-        new_servo.initialized = true;
-        new_servo.type = ServoType::POSITIONAL;
-        new_servo.setpoint_minimum = 0.0f;
-        new_servo.setpoint_maximum = 180.0f;
-        this->servos[channel] = new_servo;
-        this->setPWMBounds(channel, 500, 2500);
-    }
-}
-
-void ServoDriver::addContinuousServo(int channel){
-    if(this->isChannelNotInUse(channel)){
-        Servo new_servo;
-        new_servo.initialized = true;
-        new_servo.type = ServoType::CONTINUOUS;
-        new_servo.setpoint_minimum = -1.0f;
-        new_servo.setpoint_maximum = 1.0f;
-        this->servos[channel] = new_servo;
-        this->setPWMBounds(channel, 1100, 1900);
-    }
-}
-
-void ServoDriver::removeServo(int channel){
-    if(this->isServoChannelInUse(channel)){
-        this->servos[channel].initialized = false;
+    this->driver_board.setAllOff();
+}
+
+ServoDriver::~ServoDriver() {}
+
+void ServoDriver::registerServo(uint8_t channel, ServoType type) {
+    switch (type)
+    {
+    case ServoType::POSITIONAL:
+        servos[channel] = Servo();
+        RCLCPP_INFO(rclcpp::get_logger("ServoDriver"), "Registered positional servo on channel %d", channel);
+        setAngle(channel, 90.0f);
+        break;
+    
+    case ServoType::CONTINUOUS:
+        continuous_servos[channel] = ContinuousServo();
+        RCLCPP_INFO(rclcpp::get_logger("ServoDriver"), "Registered continuous servo on channel %d", channel);
+        // setThrottle(channel, 0.0f);
```

Comment:
COMMENT

Reply:
probably should uncomment that line. good default